ros/processor/insights_engine_result_consumer.py

```diff
@@ -189,6 +197,14 @@ def process_report(self, host, platform_metadata, reports, system_metadata, perf
                     f"{self.prefix} - Performance profile created/updated successfully for the system: {host['id']}"
                 )
                 db.session.commit()
+
+                # Trigger event for notification
+                if system_previous_state not in (SYSTEM_STATES['OPTIMIZED'], system_current_state):
+                    LOG.info(
+                        f"{self.prefix} - New recommendation event triggered for the system: {host['id']}"
+                    )
+                    new_recommendation_event(host, platform_metadata)
+
```

Comment:
Method `process_report` is getting bigger. All engine logic is mainly kept under `process_report` method. For better readability & reusability, could you try to extract out this notification part into a separate small method?